cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -248,6 +248,22 @@ open class Node @Inject constructor(private val project: Project) {
         config = config.withValue("h2port", ConfigValueFactory.fromAnyRef(h2Port))
     }
 
+    /**
+     * This section is used to configure the JDBC connection and database driver used for the node’s persistence.
+     */
+    fun dataSourceProperties(options: DataSourceProperties) {
+        config = options.addTo("dataSourceProperties", config)
+    }
+
+    /**
+     * This section is used to configure the JDBC connection and database driver used for the node’s persistence.
+     */
+    fun dataSourceProperties(configureClosure: Closure<in DataSourceProperties>) {
```

Comment:
`Closure` is a Groovy-ism. The official Kotlin equivalent these days is more like:
```kotlin
protected val dataSourceProperties: DataSourceProperties = project.objects.newInstance(DataSourceProperties::class.java)

fun dataSourceProperties(action: Action<in DataSourceProperties>) {
    action.execute(dataSourceProperties)
}
```
where `objects` is an instance of Gradle's `ObjectFactory`. Ugh, Getting Gradle DSLs right is _hard work_.